app/billing_client.go

```diff
@@ -142,15 +142,22 @@ type GetCurrentMonthUsageResponse struct {
 }
 
 func getCurrentMonthUsageResponseFromProto(response *pb.GetCurrentMonthUsageResponse) *GetCurrentMonthUsageResponse {
-	startDate := response.StartDate.AsTime()
-	endDate := response.EndDate.AsTime()
+	var startDate, endDate *time.Time
+	if response.StartDate != nil {
+		d := response.StartDate.AsTime()
```

Comment:
(q) do we need this intermediate variable? I see you're using it consistently so I imagine there's a reason why, but I would've expected this to work without.

If not, might be cleaner to just get rid of it. If so, disregard!

Reply:
This is to reflect the true nature of the variable if it's `nil`. So originally I did:
```
var startDate time.Time
if ... {
  startDate = response.StartDate.AsTime()
}
...Request{
  StartDate: &startDate
}
```
The issue with doing it this way is that `startDate`'s original value is the 0 value of `time.Time`, which is an actual time instead of `nil`. So if we pass this in as a pointer, it becomes a pointer to a zero time instead of `nil`.

Reply:
As for why I can't directly do `startDate = response.StartDate.AsTime()`, it complains:
```
invalid operation: cannot take address of response.StartDate.AsTime() (value of type time.Time)
```